Babylon/commands/api/solutions/create.py

```diff
@@ -37,35 +39,39 @@
 )
 @option("--select", "select", is_flag=True, default=True, help="Save this new solution in configuration")
 @argument("name", type=QueryType())
-@inject_context_with_resource({"api": ['url', 'organization_id'], 'acr': ['simulator_repository', 'simulator_version']})
-def create(ctx: Context, context: Any, azure_token: str, name: str, solution_file: pathlib.Path,
+@retrieve_state
+def create(state: Any, azure_token: str, organization_id: str, solution_name: str, solution_file: pathlib.Path,
            select: bool) -> CommandResponse:
     """
-    Register new solution
+    Register a new solution
     """
-    check_ascii(name)
+    state = state['state']
+    state['api']['organization_id'] = organization_id or state['api']['organization_id']
+
+    check_ascii(solution_name)
     path_file = f"{env.context_id}.{env.environ_id}.solution.yaml"
     t_file = solution_file or env.working_dir.payload_path / path_file
     if not t_file.exists():
         logger.error(f"No such file: '{basename(t_file)}' in .payload directory")
         sys.exit(1)
     details = env.fill_template(t_file,
                                 data={
-                                    "key": name.replace(" ", ""),
-                                    "name": name,
+                                    "key": solution_name.replace(" ", ""),
+                                    "name": solution_name,
                                     'tag': env.context_id.capitalize()
                                 })
-    response = oauth_request(f"{context['api_url']}/organizations/{context['api_organization_id']}/solutions",
-                             azure_token,
-                             type="POST",
-                             data=details)
+    spec = {"data": details}
+
+    service = SolutionService(state=state, azure_token=azure_token, spec=spec)
+    response = service.create()
+    created_solution = response.json()
+
     if response is None:
         return CommandResponse.fail()
-    solution = response.json()
     if select:
-        env.configuration.set_var(resource_id=ctx.parent.parent.command.name,
-                                  var_name="solution_id",
-                                  var_value=solution["id"])
-        logger.info(SUCCESS_CONFIG_UPDATED("solution", "solution_id"))
-    logger.info(SUCCESS_CREATED("solution", solution["id"]))
-    return CommandResponse.success(solution, verbose=True)
+        state["api"]["solution_id"] = created_solution["id"]
```

Comment:
are you sure it will work with variable named `state` ? when I try to execute the command, I get `KeyError: state`